test/integration/broker.test.js

```diff
@@ -17,29 +17,74 @@ const networkPort2 = 12362
 const networkPort3 = 12363
 const trackerPort = 12370
 
-describe('ws and wss connections', () => {
-    it('can connect to ws endpoint', async (done) => {
-        const broker = await startBroker('broker1', networkPort1, trackerPort, httpPort1, wsPort1, null, true)
-        const ws = new WebSocket(`ws://127.0.0.1:${wsPort1}/api/v1/ws`)
-        ws.on('open', async () => {
+describe('websocket server', () => {
+    let ws
+    let broker
+
+    afterEach(async () => {
+        if (ws) {
             ws.terminate()
-            await broker.close()
+        }
+        broker.close()
+    })
+
+    it('receives unencrypted connections', async (done) => {
+        broker = await startBroker('broker1', networkPort1, trackerPort, httpPort1, wsPort1, null, false)
+        ws = new WebSocket(getWsUrl(wsPort1))
+        ws.on('open', async () => {
             done()
         })
         ws.on('error', (err) => done(err))
     })
-    it('can connect to wss endpoint', async (done) => {
-        const broker = await startBroker('broker1', networkPort1, trackerPort, httpPort1, wsPort1, null, true, 'test/fixtures/key.pem', 'test/fixtures/cert.pem')
-        const ws = new WebSocket(`wss://127.0.0.1:${wsPort1}/api/v1/ws`, {
+
+    it('receives encrypted connections', async (done) => {
+        broker = await startBroker('broker1', networkPort1, trackerPort, httpPort1, wsPort1, null, false, 'test/fixtures/key.pem', 'test/fixtures/cert.pem')
+        ws = new WebSocket(getWsUrl(wsPort1, true), {
             rejectUnauthorized: false // needed to accept self-signed certificate
         })
         ws.on('open', async () => {
-            ws.terminate()
-            await broker.close()
             done()
         })
         ws.on('error', (err) => done(err))
     })
+
+    describe('rejections', () => {
+        beforeEach(async () => {
```

Comment:
It's easier to follow logic of test when `beforeEach` `afterEach` are near each other.